### PR TITLE
Add some note Components (NoteGrid, NoteGridCollection, and NotePreview)

### PR DESCRIPTION
**Pull request by andres-holguin**
This adds three components that will be used in the generation of notes. NoteGrid and NoteGridCollection serve as containers, while NotePreview is a visual component that shows some Note Data. Styling is subject to change, but the logic of each component is fairly sound. The NoteGridCard component referred to by NoteGridCollection.vue is a WIP, but can be seen behind the preview below.

image

### CODE REVIEW

File: src/main/webapp/components/NotePreview.vue [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    },
+    mounted: function() {
+      this.$parent.$on('open-preview', note => {
+        this.showPreview = true;
+      });
+    }
+  }
+</script>
+
+<style>
```
Comment (keelywan) [inline src/main/webapp/components/NotePreview.vue:74]: nit: I suggest adding scoped to the style tag in case it interferes with other styling

**Review by keelywan — APPROVED** [review]
Looks good!

File: src/main/webapp/components/NotePreview.vue (hunk 1 above)
Comment (andres-holguin) [reply src/main/webapp/components/NotePreview.vue:74]: Good catch. Unfortunately, the `<v-dialog>` wasn't behaving well to the scoped attribute so I left it out. I will eventually use vuetify components for the .report class so it won't be needed. Not sure what to do about iframe, but I also just tested if this does in fact affect other iframes outside of the component, and it does not. So `scoped` might not even be necessary at all 🤯 .

Comment (keelywan) [reply src/main/webapp/components/NotePreview.vue:74]: Oh ok that's good to know then!

**Review by Aradhya-B — APPROVED** [review]
Great work Andres, and the template should be super useful for future components!

**Review by nickKelman — APPROVED** [review]

**Pull request merged**